### ISSUE TITLE
Automatic interface/mli generation should use virtual files / temporary buffers

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [ocamllabs/vscode-ocaml-platform#982]
At the moment, if I need to quickly check whether a given file I'm looking at has any docs, the go-to seems to be <kbd>⌥</kbd><kbd>O</kbd>. This, sometimes-helpfully, _generates_ an interface if none exists.

However, about half the time, that means one has to deal with this dialog:

<img width="298" alt="Screen Shot 2022-07-05 at 2 37 15 PM" src="https://user-images.githubusercontent.com/200/177402885-bdfcddde-3e78-4617-90e0-c2bfed6c6474.png">

Generally speaking, I'd say that this should default to _not_ saving the file. (My logic being this: you either slightly-annoy people trying to check-the-file-if-it-exists, to save some annoyance on behalf of someone who _is_ trying to use VScode to generate the initial intf-file for a module that doesn't yet have one; or vice versa: annoy file-creators a little more ("wait! why didn't that save?"), at the cost of annoying file-readers a little less. I'd suppose that the former activity, reading, is much, much more common than the latter activity, creating.)

Would it be possible for the extension to, if the `.mli` doesn't already exist, use a [virtual document](https://code.visualstudio.com/api/extension-guides/virtual-documents) for this? Or, in some other way, prevent the above blocking modal from appearing on created MLIs?

**Comment by ulugbekna** [comment]
> I'd say that this should default to not saving the file 

I don't think we can configure that, if you intended to propose a change there.  

I just do `cmd + delete` to pick "don't save" (that's the default keybinding on mac to pick "cancel" out of two options) 

I agree that the workflow could be improved though.

**Comment by rgrinberg** [comment]
> if I need to quickly check whether a given file I'm looking at has any docs, the go-to seems to be ⌥O. This, sometimes-helpfully, generates an interface if none exists.

Why do we do this anyway? If the .mli doesn't exist I don't think we should automatically insert anything.

**Comment by ulugbekna** [comment]
We do that because the code action to insert the interface is not very discoverable for the user, IIRC. I don’t like this insertion too much either. I think we could distinguish switch with insertion to another keybiding shift+option+o (vs current option+o which would just do the switch)
